src/dlink.cc

```diff
@@ -78,3 +76,14 @@ dlinkDelete(dlink_node * m, dlink_list * list)
     m->next = m->prev = NULL;
 }
 
+void
+dlinkMoveToFront(dlink_node *node, dlink_list *list)
+{
+    if (node->next)
+        node->next->prev = node->prev;
+    if (node->prev)
+        node->prev->next = node->next;
+    node->prev = nullptr;
+    node->next = list->head;
+    list->head = node;
+}
```

Comment:
This new undocumented global function is buggy -- it is missing a `list->tail` update (at least). Please remove this function and go back to using dlinkDelete()/dlinkAdd() (instead of spending even more time on enhancing the dlink_list API that should not exist).

FWIW, I have never seen anybody being able to write doubly-linked list functions without errors on the first try. I am sure such superhumans exist, but we are not them ;-). Please do not make me re-review this function. I almost missed the known bug after the two scans and can miss more bugs. Please remove this function instead.